### ISSUE TITLE
need to add equality comparator

### ISSUE THREAD

**Issue by noname0310** [nalply/rbts#9]
202 line's `key !== node.key` is acts like a reference comparison for an Object.
so, It is impossible to process a case where an object of the nature of a record is used as a key.

It would be better to add comparator like `equalOp?: (a: K, b: K) => boolean` in `Tree<K, V>` constructor parameter.

```ts
/** @internal */_findNode(
  key: K,
  node: Node<K, V> = this._root,
): Node<K, V> {
  while (node.ok && key !== node.key)
//                  ^^^^^^^^^^^^^^^^ --- here is problem
    node = this._less(key, node) ? node._left : node._right
  return node
}
```

**Comment by nalply** [comment]
Right. It's true that `!==` is not exactly the perfect thing to do because
there's `lessOp`, however `lessOp` had a performance price. I am not sure
whether `eqOp` would be a better solution, I am sorry. I suggest you fork
and try out yourself.

Thanks for the report!

**Comment by noname0310** [comment]
Is there no change in this repository regarding this topic?
If so, it would be better to write down an additional explanation of this problem in the documentation.

+ [This is my suggestion.](https://github.com/noname0310/rbts/commit/6d4f9c0415aebf0fcc5ddea1e1a0d9be9b0cba64)

**Comment by nalply** [comment]
Please try to run a test case with at least a million entries where the
`equalOp` is used heavily (by calling `get()`, `set()` and `delete()` many
times. Compare the `equalOp` with the `!==` case and see whether there's a
big performance degradation. I would like to see the results of both runs.

When you do that it's easier for me to see whether `equalOp` causes
performance problems.

Have a look at Readme.md and click on the profiler output to see what I am writing about.
